src/client/clientenvironment.cpp

```diff
@@ -310,7 +310,8 @@ void ClientEnvironment::step(float dtime)
 		node_at_lplayer = m_map->getNode(p);
 
 		u16 light = getInteriorLight(node_at_lplayer, 0, m_client->ndef());
-		lplayer->light_color = encode_light(light, 0); // this transfers light.alpha
+		lplayer->light_color = encode_light(light, 0,
+			m_client->getEnv().getLocalPlayer()->getLighting().ambient_light.luminance); // this transfers light.alpha
```

Comment:
Forgive my ignorance, but isn't `m_client->getEnv().getLocalPlayer()` the same as `lplayer`?

Reply:
Oh, actually yes. 149th line defines it.